src/app/zap-templates/zcl/data-model/chip/joint-fabric-datastore-cluster.xml

```diff
@@ -0,0 +1,298 @@
+<?xml version="1.0"?>
+<!--
+Copyright (c) 2024 Project CHIP Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<configurator>
+  <domain name="CHIP"/>
+
+  <enum name="DatastoreStateEnum" type="enum8">
+    <cluster code="0x0752"/>
+    <item name="Pending" value="0x00"/>
+    <item name="Committed" value="0x01"/>
+    <item name="DeletePending" value="0x02"/>
+    <item name="CommitFailed" value="0x03"/>
+  </enum>
+
+  <enum name="FailureCodeEnum" type="enum8">
```

Comment:
This is not in the spec.

Reply:
Which version of the spec are you referring to?

Reply:
> Which version of the spec are you referring to?

SHA 589d3bbd91653e7ae1e48107fc86b08c8fbee8ec

Reply:
Regarding _FailureCodeEnum_ I agree with @bzbarsky-apple - this field doesn't exist in the current spec/any un-merged spec-PR. What exists, is the _FailureCode_, which is a field of the _DatastoreStatusEntryStruct_.

What confuses me is why the current PR makes the assumption that only _OK_ is a possible value for this enum. I was reading the spec text around the _FailureCode_ and I think this field should be of [_status_ type](https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/data_model/Data-Model.adoc#20221-status-code-type) instead of _enum_ type.

Reply:
correct